add bt scan permission to manifest and init routine of DalvikBleService

macgibbon opened the pull request:
Android API level 3 has a new permission for BLE peripherals as described  [here](https://developer.android.com/guide/topics/connectivity/bluetooth/permissions).

macgibbon committed:
add bt scan permission to manifest and init routine of DalvikBleService

jperedadnr commented:
I take that you mean Android 31+ ?

In that case, shouldn't we add a check on the version and only apply the changes to 

Can you update the copyright year to 2023, and sign the CLA if you haven't done it yet?

jperedadnr commented on modules/ble/src/main/resources/META-INF/substrate/dalvik/AndroidManifest.xml:
```diff
@@ -3,6 +3,7 @@
     <uses-permission android:name="android.permission.BLUETOOTH" />
     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
-    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
```
Don't you need to add `android:maxSdkVersion="30"` to legacy permissions like in [here](https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#declare-android12-or-higher)?

jperedadnr commented:
@macgibbon are you willing to address the feedback? In case that you can't, no problem, I'll do a PR myself with the required changes, in order to fix this issue and release Attach 4.0.18 with this and the recent changes needed for Android 31

jperedadnr commented:
closed by #357

macgibbon commented:
I apologize for not addressing the feedback, I didn't have notifications set up properly and sort of moved on to other activities with  the onset of good weather.  Thanks for implementing the fixes!
